pkg/cmd/infra/router/template.go

```diff
@@ -251,6 +251,9 @@ func (o *TemplateRouterOptions) Run() error {
 		if len(o.StatsUsername) == 0 || len(o.StatsPassword) == 0 {
 			glog.Warningf("Metrics were requested but no username or password has been provided - the metrics endpoint will not be accessible to prevent accidental security breaches")
 		}
+		if statsPort == 0 {
+			return fmt.Errorf("When type is haprocy-router, STATS_PORT must not be 0")
```

Comment:
lower case, typo, and not the right flag:

```
when --metrics-type is haproxy --stats-port may not be set to 0
```

Reply:
@smarterclayton This message comes up when you run oc adm router --stats-port=0
--metrics-type is an invalid 'oc adm router' parameter so message will just confuse the user.
Made the requested changes.